### PR TITLE
Integrate Sentry

### PR DESCRIPTION
**Pull request by disq**
What more do we need?

### CODE REVIEW

**Review by spangenberg — COMMENTED** [review]
LGTM

**Review by yevgenypats — DISMISSED** [review]
Where is the opt-out flag?

**Comment by disq** [comment]
> Where is the opt-out flag?

oops. added.

**Review by spangenberg — APPROVED** [review]

File: pkg/client/client.go [hunk 1]
```diff
@@ -1050,3 +1052,15 @@ func collectFetchSummaryStats(span otrace.Span, fetchSummaries map[string]Provid
...
 }
+
+// reportFetchSummaryStats reads provided fetch summaries and sends them to sentry
+func reportFetchSummaryStats(fetchSummaries map[string]ProviderFetchSummary) {
+	for _, ps := range fetchSummaries {
+		for _, e := range ps.Diagnostics() {
+			if e.Severity() == diag.ERROR {
+				d := e.Description()
+				sentry.CaptureException(fmt.Errorf("%s: %s: %s: %s: %s", ps.ProviderName, e.Type().String(), d.Resource, d.Summary, d.Detail))
+			}
```
Comment (roneli) [inline pkg/client/client.go:1063]:
```suggestion
			if e.Severity() != diag.ERROR {
			  continue
			}
				d := e.Description()
				sentry.CaptureException(fmt.Errorf("%s: %s: %s: %s: %s", ps.ProviderName, e.Type().String(), d.Resource, d.Summary, d.Detail))
```

File: pkg/client/client.go [hunk 2]
```diff
@@ -518,6 +519,7 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
 	}
 
 	collectFetchSummaryStats(otrace.SpanFromContext(ctx), response.ProviderFetchSummary)
+	reportFetchSummaryStats(response.ProviderFetchSummary)
```
Comment (roneli) [inline pkg/client/client.go:522]:
```suggestion
	reportFetchSummaryErrors(response.ProviderFetchSummary)
```

Comment (roneli) [inline pkg/client/client.go:522]: I think you could probably combine both methods to reduce the looping, wdyt?

**Review by roneli — CHANGES_REQUESTED** [review]
Would also suggest to do some manual tests like running AWS provider with bad credentials

Also write some tests for example test provider and see that errors get captured and sent.

For AWS provider I would test that the no creds are leaked, they shouldn't but just to double check.

File: internal/telemetry/error.go [hunk 3]
```diff
@@ -11,6 +12,8 @@ func RecordError(span otrace.Span, err error, opts ...otrace.EventOption) {
 		return
 	}
 
+	sentry.CaptureException(err)
+
 	//  TODO for fetch get table name / error type
```
Comment (roneli) [inline internal/telemetry/error.go:17]: we do this no with resource name etc' so we can remove this TODO

**Review by roneli — APPROVED** [review]
LGTM, minor nit

**Pull request merged**